repos/system_upgrade/common/actors/trackcustommodifications/libraries/trackcustommodifications.py

```diff
@@ -0,0 +1,92 @@
+    if not actor:
+        # Assuming here we are dealing with a library, so let's discover actor file and deduce actor name from it
+        # actor is expected to be found under ../../actor.py
+        assumed_actor_file = os.path.join(a_file.split('libraries')[0], 'actor.py')
+        if not os.path.exists(assumed_actor_file):
+            # Nothing more we can do - no actor name mapping, return ''
+            return ''
+        return deduce_actor_name(assumed_actor_file)
+    return actor.name
+
+
+def _run_command(cmd, warning_to_log, checked=True):
+    """
+    A helper that executes a command and returns a result or raises StopActorExecution.
+    Upon success results will contain a list with line-by-line output returned by the command.
+    """
+    try:
+        res = run(cmd, checked=checked)
+        output = res['stdout'].strip()
+        if not output:
+            return []
+        return output.split('\n')
+    except CalledProcessError:
+        api.current_logger().warning(warning_to_log)
+        raise StopActorExecution()
+
+
+def check_for_modifications(dirs=None):
+    """
+    This will return a tuple (changes, custom) is case any untypical files or changes to shipped leapp files are
+    discovered.
+    (None, None) means that no modifications have been found.
+    """
+    dirs = dirs if dirs else DIRS_TO_SCAN
+    rpms = _get_rpms_to_check()
+    source_of_truth = []
+    leapp_files = []
+    # Let's collect data about what should have been installed from rpm
+    for rpm in rpms:
+        res = _run_command(['rpm', '-ql', rpm], 'Could not get a list of installed files from rpm {}'.format(rpm))
+        source_of_truth.extend(res)
+    # Let's collect data about what's really on the system
+    for directory in dirs:
+        res = _run_command(['find', directory, '-type', 'f'],
+                           'Could not get a list of leapp files from {}'.format(directory))
+        leapp_files.extend(res)
+    # Let's check for unexpected additions
+    custom_files = sorted(set(leapp_files) - set(source_of_truth))
+    # Now let's check for modifications
+    modified_files = []
+    for rpm in rpms:
+        res = _run_command(['rpm', '-V', rpm], 'Could not check authenticity of the files from {}'.format(rpm),
+                           # NOTE(ivasilev) check is False here as in case of any changes found exit code will be 1
+                           checked=False)
+        if res:
+            api.current_logger().warning('Modifications to leapp files detected!\n%s', res)
+            modified_files.extend([tuple(x.split()) for x in res])
+    return ([CustomModifications(actor_name=deduce_actor_name(f[1]), filename=f[1], type='modified',
+                                 rpm_checks_str=f[0])
+             for f in modified_files],
+            [CustomModifications(actor_name=deduce_actor_name(f), filename=f, type='custom') for f in custom_files])
```

Comment:
well, if we do not do the change (and there are not any such plans and reasons for it) I think we should stick to current requirements of the leapp framework. I could say that from the same point this unlimited check does not bring a benefit, as we could say that we could modify the definition of actors and in such a case this will be invalid also. The chance for it is the very same as for a case to drop the requirement in the framework to define an actor inside actor.py file.

In case we do not find agreement here, I think that in such a case we should use the framework's functions directly to do the scan for us. Or we could create instead a ticket to work on it later. Currently I do not see, based on above, any benefit to make our lifes harder in this case. It can be put on a cabal discussion also.

Reply:
I hear a lot of "if we do something in the future then this will need to be changed" here :( Putting the scanning in the framework can be done but then this PR might not  be getting in this week, as it will be a big tricky to test change.
If the requirement is to inform the operator about changes in particular actors, then reporting code changes in actor library is imho *a must*. So I see a huge benefit that will make the lives of leapp operators easier.

Reply:
My concern is not about shared libraries (those will be reported as not belonging to any actor), it's about *a particular actor libraries* - python modules that reside under thisactor/libraries directory and are precisely mapped 1-1 to this particular actor. If we are reporting actor names in case of changes - this is precisely what should be reported and I see no benefit of getting rid of this.

Reply:
Updated the code with tracking actor files + covered with unit tests. Love the idea, thanks a lot!